python/foglamp/tasks/purge/purge.py

```diff
@@ -106,39 +106,48 @@ def purge_data(self, config):
         result = self._storage.query_tbl_with_payload("streams", payload)
         last_id = result["rows"][0]["min_last_object"] if result["count"] == 1 else 0
 
-
         flag = "purge" if config['retainUnsent']['value'] == "False" else "retain"
-        if config['age']['value'] != 0:
-            result = self._readings_storage.purge(age=config['age']['value'], sent_id=last_id, flag=flag)
-
-            if "message" in result.keys() and "409 Conflict" in result["message"]:
-                self._logger.error("Purge failed: %s", result["message"])
-            else:
-                total_count = result['readings']
-                total_rows_removed = result['removed']
-                unsent_rows_removed = result['unsentPurged']
-                unsent_retained = result['unsentRetained']
-
-        if config['size']['value'] != 0:
-            result = self._readings_storage.purge(size=config['size']['value'], sent_id=last_id, flag=flag)
-
-            if "message" in result.keys() and "409 Conflict" in result["message"]:
-                self._logger.error("Purge failed: %s", result["message"])
-            else:
-                total_count += result['readings']
-                total_rows_removed += result['removed']
-                unsent_rows_removed += result['unsentPurged']
-                unsent_retained += result['unsentRetained']
+        try:
+            if int(config['age']['value']) != 0:
+
+                result = self._readings_storage.purge(age=config['age']['value'], sent_id=last_id, flag=flag)
+
+                if "message" in result.keys() and "409 Conflict" in result["message"]:
+                    self._logger.error("Purge failed: %s", result["message"])
+                else:
+                    total_count = result['readings']
+                    total_rows_removed = result['removed']
+                    unsent_rows_removed = result['unsentPurged']
+                    unsent_retained = result['unsentRetained']
+        except ValueError:
+            self._logger.error("Configuration item age {} should be integer!".format(config['age']['value']))
+            raise ValueError
```

Comment:
What will happen, If any other exception occurred?

Reply:
No other exception raised till date, if it occurs, let it fail post that we can add a block to handle that too. FYI, point of this pull is just to add unit test and fix FOGL-1090 and exception block added to capture that only.

Reply:
`run` def already has  generic exception handler block which calls purge_data. Hence not needed!